### PR TITLE
python3Packages.connexion: 2.7.0 -> 2.9.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 2.9.0

Change log: https://github.com/zalando/connexion/releases/tag/2.9.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 133333` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.connexion</li>
    <li>python39Packages.connexion</li>
  </ul>
</details>

File: pkgs/development/python-modules/connexion/default.nix [hunk 1]
```diff
@@ -54,6 +55,15 @@ buildPythonPackage rec {
     testfixtures
   ];
 
+  patches = [
+    # No minor release for later versions, https://github.com/zalando/connexion/pull/1402
+    (fetchpatch {
+      name = "switch-to-poetry-core.patch";
+      url = "https://github.com/zalando/connexion/commit/4a225d554d915fca17829652b7cb8fe119e14b37.patch";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/connexion/default.nix:62]: That patch is bumping flask and werkzeug.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133333` at e998ab29 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.connexion</li>
    <li>python39Packages.connexion</li>
  </ul>
</details>

**Comment by risicle** [comment]
Only concern is that the patch isn't in upstream's master.

Otherwise, `nixpkgs-review` happy on macos 10.15 & nixos x86_64.

**Comment by fabaff** [comment]
> Only concern is that the patch isn't in upstream's master.

Their workflow is no clear to be but I hope that they are merging the branch in question in main/master before releasing.

**Pull request merged**